src/map.js

```diff
@@ -59,13 +59,15 @@ class Map {
 	isTileOccupied(x, y) {
 		let entity = this.getEntityAt(x, y);
 
-		return entity !== undefined;
+		return entity !== undefined && entity.isAlive();
```

Comment:
We may not want to tie in an `isAlive` here, but is that part of the problem we were facing?